flink-tests/src/test/java/org/apache/flink/test/recovery/JobManagerHAProcessFailureBatchRecoveryITCase.java

```diff
@@ -141,15 +151,14 @@ public JobManagerHAProcessFailureBatchRecoveryITCase(ExecutionMode executionMode
 	 */
 	private void testJobManagerFailure(String zkQuorum, final File coordinateDir, final File zookeeperStoragePath) throws Exception {
 		Configuration config = new Configuration();
-		config.setString(CoreOptions.MODE, CoreOptions.LEGACY_MODE);
 		config.setString(HighAvailabilityOptions.HA_MODE, "ZOOKEEPER");
 		config.setString(HighAvailabilityOptions.HA_ZOOKEEPER_QUORUM, zkQuorum);
 		config.setString(HighAvailabilityOptions.HA_STORAGE_PATH, zookeeperStoragePath.getAbsolutePath());
 
 		ExecutionEnvironment env = ExecutionEnvironment.createRemoteEnvironment(
 				"leader", 1, config);
 		env.setParallelism(PARALLELISM);
-		env.setNumberOfExecutionRetries(1);
+		env.setRestartStrategy(RestartStrategies.fixedDelayRestart(1, 0L));
 		env.getConfig().setExecutionMode(executionMode);
```

Comment:
Hi @tillrohrmann , I want to know how `ExecutionMode` affect `DataExchangeMode` here. If I understand it correctly, this test case tests both `BATCH` and `PIPELINED` `DataExchangeMode`. But only `.rebalance()` here, I think I have to learn which type `rebalance` belong to.

Further, maybe I can port `JobManagerFailsITCase` based on this case, even some are covered.

Reply:
The `ExecutionMode` defines the data exchange between producers and consumers. When using `BATCH`, then the producers (e.g. input for a reduce operation) first need to finish before the consumer is started. In `PIPELINED` mode, the consumer will start as soon as a single producer has produced some consumable data.

The test job contains a shuffle operation (between mapper and reducer) which is affected by this setting.